### ISSUE TITLE
ec2_transit_gateway_vpc_attachment: add support for "enable_appliance_mode"

### ISSUE THREAD

**Issue by nckbnv** [hashicorp/terraform-provider-aws#16161]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS has just released a new type of load balancer named Gateway Load Balancer. 
It supports three design options:
* North-South inspection (VPC to/from Internet) via GLWB and IGW (Internet Gateway) covered by #16129 and released in provider version 3.15
* North-South inspection (VPC to/from Internet) via GLWB and TGW (Transit Gateway)
* East-West (Inter-VPC traffic inspection) via GLWB and TGW

In order to prevent asymmetric routing between GLWB and TGW (so one session could always hit the proper firewall appliance) you need to enable Appliance mode in the transit gateway vpc attachment ([as described in the Appliance VPC documentation](https://docs.aws.amazon.com/vpc/latest/tgw/transit-gateway-appliance-scenario.html#transit-gateway-appliance-routing))

This feature enhancement request is opened based on a suggestion by @bflad made in [16129.](https://github.com/hashicorp/terraform-provider-aws/issues/16129#issuecomment-725573275_) 

### New or Affected Resource(s)

* ec2_transit_gateway_vpc_attachment

### Potential Terraform Configuration

```hcl
resource "aws_ec2_transit_gateway_vpc_attachment" "example" {
  subnet_ids         = [aws_subnet.example.id]
  transit_gateway_id = aws_ec2_transit_gateway.example.id
  vpc_id             = aws_vpc.example.id
  appliance_mode = true
}
```

### References

* #16129

**Comment by zhecho** [comment]
+1

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.15.0 of the Terraform AWS Provider, today or tomorrow. Thanks to @philnichol for the implementation. 👍
